core/breakpoints/breakpoint.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Elementor\Core\Breakpoints;
+
+use Elementor\Plugin;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Breakpoint {
+
+	const BREAKPOINT_OPTION_PREFIX = 'viewport_';
+
+	private $name;
+	private $default_value;
+	private $db_key;
+	private $value;
+	private $direction = 'max';
+	private $is_enabled = false;
+
+	/**
+	 * Get Name
+	 *
+	 * @since 3.1.0
+	 *
+	 * @return string
+	 */
+	public function get_name() {
+		return $this->name;
+	}
+
+	/**
+	 * Get Initial Config
+	 *
+	 * @since 3.1.0
+	 *
+	 * @return array Initial Config
+	 */
+	public function get_config() {
+		return [
+			'value' => $this->get_value(),
+			'direction' => $this->direction,
+			'is_enabled' => $this->is_enabled(),
+		];
+	}
+
+	/**
+	 * Is Enabled
+	 *
+	 * Check if the breakpoint is enabled or not. The breakpoint instance receives this data from
+	 * the Breakpoints Manager.
+	 *
+	 * @return bool $this->is_enabled
+	 */
+	public function is_enabled() {
+		return $this->is_enabled;
+	}
+
+	/**
+	 * Get Value
+	 *
+	 * Retrieve the saved breakpoint value.
+	 *
+	 * @return array|mixed|string
+	 */
+	public function get_value() {
+		if ( ! $this->value ) {
+			$value_from_db = Plugin::$instance->kits_manager->get_current_settings( $this->db_key );
+
+			$this->value = isset( $value_from_db ) ? $value_from_db : $this->default_value;
```

Comment:
Add a method `is_custom` to check if a breakpoint has a custom value

Reply:
Done.